### PR TITLE
ASV: clean/simplify setup methods

### PR DESCRIPTION
**Pull request by lukemanley**
Trying to make some of the setup methods a bit more efficient. Might(?) save around a minute total on the ASV job. 

e.g.

```
import time
from itertools import product
from asv_bench.benchmarks.series_methods import NanOps

param_sets = list(product(*NanOps.params))

t = time.time()
for params in param_sets:
    try:
        NanOps().setup(*params)
    except NotImplementedError:
        pass
print(time.time() - t)

# 11.87  -> main
# 1.90   -> PR
```

### CODE REVIEW

File: asv_bench/benchmarks/index_object.py [hunk 1]
```diff
@@ -161,9 +161,7 @@ def setup(self, dtype):
         self.sorted = self.idx.sort_values()
         half = N // 2
         self.non_unique = self.idx[:half].append(self.idx[:half])
-        self.non_unique_sorted = (
-            self.sorted[:half].append(self.sorted[:half]).sort_values()
-        )
+        self.non_unique_sorted = self.sorted[:half].repeat(2)
```
Comment (WillAyd) [inline asv_bench/benchmarks/index_object.py:164]: Is the sort_values here no longer required?

**Review by WillAyd — COMMENTED** [review]
I think a lot of these are reasonable - thanks for the cleanup

File: asv_bench/benchmarks/index_object.py (hunk 1 above)
Comment (lukemanley) [reply asv_bench/benchmarks/index_object.py:164]: Thats correct, it is no longer required. The array is already sorted so using `repeat` here will maintain the sort.

**Review by WillAyd — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @lukemanley

**Pull request merged**